packages/p2p/src/pages/app.jsx

```diff
@@ -122,6 +120,15 @@ const App = () => {
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    // Redirect to /p2p/buy-sell if user navigates to /p2p without a subroute
+    React.useEffect(() => {
+        if (/\/p2p$/.test(location.pathname) || location.pathname === '/cashier/p2p/') {
+            history.push(routes.p2p_buy_sell);
+            general_store.setActiveIndex(0);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [location.pathname]);
+
```

Comment:
the main reason why we got the blank page when clicking the p2p tab twice was cause the routing was only handled in the first time p2p page is mounted in the useEffect, but since clicking the tab resets the path back to `cashier/p2p` and not `cashier/p2p/buy-sell` but there's no actual route for that. So, instead opt for useEffect to listen into location.pathname, and specifically check if we keep routing to just `p2p` or `p2p/` where no trailing subroute is there, and always default load buy sell page.